### PR TITLE
Lvjavj 27 Address classificator functionality (BackEnd only)

### PR DESCRIPTION
**Pull request by YuriPushchalo**
Region, City, Street entity classes are created
Region, City, Street repositories are created
Appropriate Liquibase changeSets are added/modified.
SQL ImportScript  is modified.
AddressService is created/implemented.
AddressController is created.

For additional info see http://ssu-jira.softserveinc.com/browse/LVJAVJ-27

### CODE REVIEW

File: persistence/src/main/resources/db/changelog/City/create-city-table.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+              "referencedColumnNames": "region_id",
+              "referencedTableName": "regions"
+            }
+          }
+        ]
+      }
+    }
+    
+  ]
+}
```
Comment (KostyantynPanchenko) [inline persistence/src/main/resources/db/changelog/City/create-city-table.json]: Add new line at the end of file

File: persistence/src/main/resources/db/changelog/Street/create-street-table.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+              "referencedColumnNames": "city_id",
+              "referencedTableName": "cities"
+            }
+          }
+        ]
+      }
+    }
+    
+  ]
+}
```
Comment (KostyantynPanchenko) [inline persistence/src/main/resources/db/changelog/Street/create-street-table.json]: Add new line at the end of file

File: persistence/src/main/resources/db/changelog/region/create-region-table.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                }
+              ],
+              "tableName": "regions"
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
```
Comment (KostyantynPanchenko) [inline persistence/src/main/resources/db/changelog/region/create-region-table.json]: Add new line at the end of file

File: service/src/main/java/com/softserve/osbb/service/impl/AddressServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * Created by YuriPushchalo on 16.11.2016
+ */
+
```
Comment (KostyantynPanchenko) [inline service/src/main/java/com/softserve/osbb/service/impl/AddressServiceImpl.java]: Delete blank line after class java doc

File: service/src/main/java/com/softserve/osbb/service/AddressService.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import com.softserve.osbb.model.City;
+import com.softserve.osbb.model.Street;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Created by YuriPushchalo on 16.11.2016
+ */
+
```
Comment (KostyantynPanchenko) [inline service/src/main/java/com/softserve/osbb/service/AddressService.java]: Delete blank line after class java doc

File: service/src/main/java/com/softserve/osbb/service/impl/AddressServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	RegionRepository regionRepository;
+
+	@Transactional(readOnly = true, propagation = Propagation.SUPPORTS)
+	@Override
+	public List<Region> getAllRegion() {
+		return regionRepository.findAll();
+	}
+
+	@Autowired
+	CityRepository cityRepository;
```
Comment (KostyantynPanchenko) [inline service/src/main/java/com/softserve/osbb/service/impl/AddressServiceImpl.java]: Move to the top

File: service/src/main/java/com/softserve/osbb/service/impl/AddressServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+	@Transactional(readOnly = true, propagation = Propagation.SUPPORTS)
+	@Override
+	public List<City> getAllCitiesOfRegion(Integer regionId) {
+		return cityRepository.findByRegion(regionRepository.findByRegionId(regionId));
+	}
+
+	@Autowired
+	StreetRepository streetRepository;
```
Comment (KostyantynPanchenko) [inline service/src/main/java/com/softserve/osbb/service/impl/AddressServiceImpl.java]: Move to the top

**Review by KostyantynPanchenko — CHANGES_REQUESTED** [review]
Approved. Please make minor styling changes.

**Comment by YuriPushchalo** [comment]
Changes is done.

**Review by KostyantynPanchenko — APPROVED** [review]
Approved

**Pull request merged**